src/chain/bitcoind.rs

```diff
@@ -354,6 +354,14 @@ impl BitcoindChainSource {
 					now.elapsed().unwrap().as_millis()
 				);
 				*self.latest_chain_tip.write().unwrap() = Some(tip);
+
+				periodically_archive_fully_resolved_monitors(
```

Comment:
Context question: channel manager already knows the best block and I assume it is updated somehow, but still we need an external trigger here to archive? It seems responsibility is now with two units instead of one.